### PR TITLE
feat: added possibility to filter experiments for models

### PR DESCRIPTION
**Pull request by abhilash-kumar-nair**

### CODE REVIEW

File: tests/impact/client/sal/test_workspace_service.py [hunk 1]
```diff
@@ -161,6 +161,12 @@ def test_get_experiments(self, get_all_experiments):
                 ]
             }
         }
+        assert get_all_experiments.adapter.called
+        get_exps_call = get_all_experiments.adapter.request_history[0]
+        assert (
+            f'http://mock-impact.com/api/workspaces/workspace_1/experiments?classPath='
```
Comment (efredriksson-modelon) [inline tests/impact/client/sal/test_workspace_service.py:167]: This does not look right, `?classPath=` should not be in the url when not asking for any specific classpath.

File: tests/impact/client/sal/test_workspace_service.py [hunk 2]
```diff
@@ -161,6 +161,12 @@ def test_get_experiments(self, get_all_experiments):
                 ]
             }
         }
+        assert get_all_experiments.adapter.called
```
Comment (efredriksson-modelon) [inline tests/impact/client/sal/test_workspace_service.py:164]: Can you also add a test where `experiments_get` is called with a class_path?

**Review by efredriksson-modelon — CHANGES_REQUESTED** [review]

**Comment by abhilash-kumar-nair** [comment]
Thanks for the review @efredriksson-modelon . I have made the suggested changes!

**Comment by abhilash-kumar-nair** [comment]
> From the discussion I still think it is unclear what the use case is. Depending on this you could either do it like this or have it on the model class:
> 
> ```
> model = workspace.get_model(...)
> model.get_experiments()
> ```
> 
> If we could have some end-user script on how this feature would be used I think it would become more apparent which solution is most intuitive.

I agree. Thanks for the ideas Emil. I have now made this change. I am not sure, how much will this be helpful for the user.

**Comment by iakovn** [comment]
> From the discussion I still think it is unclear what the use case is. Depending on this you could either do it like this or have it on the model class:
> ```
> model = workspace.get_model(...)
> model.get_experiments()
> ```
> 
> If we could have some end-user script on how this feature would be used I think it would become more apparent which solution is most intuitive.

Let me try to explain the use case a bit more. We currently only have "full forward flow":
```
model = workspace.get_model(...)
experiment_definition = model.new_experiment_definition(dynamic)
exp = workspace.execute(experiment_definition).wait()
do_postprocessing(exp)
```
The missing part is "restarted" flow:
```
model = workspace.get_model(...)
experiment_definition = model.new_experiment_definition(dynamic)
exps = workspace.get_experiments(class_name)
exp = None
for e  in exps:
   if is_this_matching_my_analysis(e, experiment_definition):
      exp = e
      break
if not exp:
    exp = workspace.execute(experiment_definition).wait()
do_postprocessing(exp)
```

**Comment by abhilash-kumar-nair** [comment]
> get_experiments

Thanks for the comments @iakovn . I have now moved the method to the workspace entity.

**Comment by abhilash-kumar-nair** [comment]
> From the discussion I still think it is unclear what the use case is. Depending on this you could either do it like this or have it on the model class:
> 
> ```
> model = workspace.get_model(...)
> model.get_experiments()
> ```
> 
> If we could have some end-user script on how this feature would be used I think it would become more apparent which solution is most intuitive.

Hej @efredriksson-modelon , I was not able to implement the above suggestion due to a circular import(See https://jenkins.modelon.com/internal/blue/organizations/jenkins/WAMSTeam%2Fimpact-client-python/detail/experiment-for-model/8/pipeline/ for the failure), so I have now moved the functionality now to the workspace entity.

**Review by iakovn — APPROVED** [review]
I'm find with this implementation. I do not have a strong opinion if we should have `model.get_experiments()`, `workspace.get_experiments(class_path)` or both as it seems that different flows maybe relevant depending on the specific workflow.

**Pull request merged**